[google_workspace] Add data studio event type

navnit-elastic opened the pull request:
<!-- Type of change
Please label this PR with one of the following labels, depending on the scope of your change:
- Bug
- Enhancement
- Breaking change
- Deprecation
-->

## Proposed commit message
```
google_workspace: add support for data studio event type as a new data stream

This adds support for the Data Studio audit event type as a new data stream to
enhance the overall visibility of data in the Google Workspace integration.
```

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).
- [x] I have verified that any added dashboard complies with Kibana's [Dashboard good practices](https://docs.elastic.dev/ux-guidelines/data-viz/dashboard-good-practices) 

## Author's Checklist

<!-- Recommended
Add a checklist of things that are required to be reviewed in order to have the PR approved
-->
- [ ] 

## How to test this PR locally

1. Clone integrations repo.
2. Install elastic package locally.
3. Start elastic stack using elastic-package.
4. Move to integrations/packages/google_workspace directory.
5. Run the following command to run tests. 
> elastic-package test

```
--- Test results for package: google_workspace - START ---
╭──────────────────┬─────────────┬───────────┬─────────────────────────────────────────────────┬────────┬──────────────╮
│ PACKAGE          │ DATA STREAM │ TEST TYPE │ TEST NAME                                       │ RESULT │ TIME ELAPSED │
├──────────────────┼─────────────┼───────────┼─────────────────────────────────────────────────┼────────┼──────────────┤
│ google_workspace │ data_studio │ pipeline  │ (ingest pipeline warnings test-data-studio.log) │ PASS   │ 321.130301ms │
│ google_workspace │ data_studio │ pipeline  │ test-data-studio.log                            │ PASS   │ 392.514786ms │
╰──────────────────┴─────────────┴───────────┴─────────────────────────────────────────────────┴────────┴──────────────╯
--- Test results for package: google_workspace - END   ---
Done
```
```
--- Test results for package: google_workspace - START ---
╭──────────────────┬─────────────┬───────────┬───────────┬────────┬───────────────╮
│ PACKAGE          │ DATA STREAM │ TEST TYPE │ TEST NAME │ RESULT │  TIME ELAPSED │
├──────────────────┼─────────────┼───────────┼───────────┼────────┼───────────────┤
│ google_workspace │ data_studio │ system    │ default   │ PASS   │ 34.272091435s │
╰──────────────────┴─────────────┴───────────┴───────────┴────────┴───────────────╯
--- Test results for package: google_workspace - END   ---
Done
```
## Related issues

- Closes #13340 

## Screenshots

![google-workspace-data-studio-screenshot](https://github.com/user-attachments/assets/8b74a36a-e915-415e-a12d-d335713c87e2)

elasticmachine commented:
Pinging @elastic/security-service-integrations (Team:Security-Service Integrations)

kcreddy commented on packages/google_workspace/data_stream/data_studio/manifest.yml:
```diff
@@ -0,0 +1,103 @@
+        multi: false
+        required: true
+        show_user: false
+        description: Page size for the response of the Google Workspace Reporting API. Maximum page size can be 1000.
```
I don't see the documented max page size in the API doc referenced in README: https://developers.google.com/workspace/admin/reports/v1/appendix/activity/data-studio

Could you point me where its derived from?

kcreddy commented on packages/google_workspace/data_stream/data_studio/manifest.yml:
```diff
@@ -0,0 +1,103 @@
+        required: true
+        show_user: true
+        default: 2h
+      - name: page_size
```
Lets rename this to `batch_size`. I think most integrations call it that way.

kcreddy commented on packages/google_workspace/data_stream/data_studio/agent/stream/cel.yml.hbs:
```diff
@@ -0,0 +1,100 @@
+           "last_timestamp": state.end_time,
+         },
+        "want_more": has(body.nextPageToken),
+        ?"next_page_token": body.?nextPageToken,
```
Move this inside cursor, just like chrome: https://github.com/elastic/integrations/blob/main/packages/google_workspace/data_stream/chrome/agent/stream/cel.yml.hbs#L59

kcreddy commented on packages/google_workspace/data_stream/data_studio/agent/stream/cel.yml.hbs:
```diff
@@ -0,0 +1,100 @@
+          "error": {
+            "code": string(resp.StatusCode),
+            "id": string(resp.Status),
+            "message": "GET:"+(
```
Follow this message: https://github.com/elastic/integrations/blob/main/packages/proofpoint_itm/data_stream/report/agent/stream/cel.yml.hbs#L55

kcreddy commented on packages/google_workspace/img/google-workspace-data-studio-screenshot.png:
Add a navigation panel, just like chrome dashboard: https://github.com/elastic/integrations/blob/main/packages/google_workspace/img/google-workspace-chrome-screenshot.png

navnit-elastic replied:
I haven't added it, as it becomes deprecated when a new data stream is added.

navnit-elastic commented:
Hi @kcreddy, I've addressed above comments and updated the PR accordingly, the changes are system tested. Please let me if we’d like to add the navigation panel to the dashboard.

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/24188)
* Commit: 590226cd41ba4ad1f3dc01a1d46eaaa09f85192e

### Failed CI Steps
* [Check integrations google_workspace](https://buildkite.com/elastic/integrations/builds/24188#0195f553-b5ee-4b76-af58-017427096272)

### History
* :green_heart: [Build #24104](https://buildkite.com/elastic/integrations/builds/24104) succeeded b2db8ff0bb0255d51ce8573d2c2ed0de87d7c7e5

cc @navnit-elastic

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/24188","number":24188,"commit":"590226cd41ba4ad1f3dc01a1d46eaaa09f85192e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/24104","number":24104,"commit":"b2db8ff0bb0255d51ce8573d2c2ed0de87d7c7e5"}],"number":24188}
buildkite-pr-comment-->

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/24189)
* Commit: 559379f73866f69ecb4f349a0e1a3d5d6f536f37

### History
* :broken_heart: [Build #24188](https://buildkite.com/elastic/integrations/builds/24188) failed 590226cd41ba4ad1f3dc01a1d46eaaa09f85192e
* :green_heart: [Build #24104](https://buildkite.com/elastic/integrations/builds/24104) succeeded b2db8ff0bb0255d51ce8573d2c2ed0de87d7c7e5

cc @navnit-elastic

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/24189","number":24189,"commit":"559379f73866f69ecb4f349a0e1a3d5d6f536f37"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/24188","number":24188,"commit":"590226cd41ba4ad1f3dc01a1d46eaaa09f85192e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/24104","number":24104,"commit":"b2db8ff0bb0255d51ce8573d2c2ed0de87d7c7e5"}],"number":24189}
buildkite-pr-comment-->

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/24204)
* Commit: db506d4d7107277a204422c54ab379e42d1ae610

### History
* :green_heart: [Build #24189](https://buildkite.com/elastic/integrations/builds/24189) succeeded 559379f73866f69ecb4f349a0e1a3d5d6f536f37
* :broken_heart: [Build #24188](https://buildkite.com/elastic/integrations/builds/24188) failed 590226cd41ba4ad1f3dc01a1d46eaaa09f85192e
* :green_heart: [Build #24104](https://buildkite.com/elastic/integrations/builds/24104) succeeded b2db8ff0bb0255d51ce8573d2c2ed0de87d7c7e5

cc @navnit-elastic

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/24204","number":24204,"commit":"db506d4d7107277a204422c54ab379e42d1ae610"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/24189","number":24189,"commit":"559379f73866f69ecb4f349a0e1a3d5d6f536f37"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/24188","number":24188,"commit":"590226cd41ba4ad1f3dc01a1d46eaaa09f85192e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/24104","number":24104,"commit":"b2db8ff0bb0255d51ce8573d2c2ed0de87d7c7e5"}],"number":24204}
buildkite-pr-comment-->

kcreddy commented:
> Hi @kcreddy, I've addressed above comments and updated the PR accordingly, the changes are system tested. Please let me if we’d like to add the navigation panel to the dashboard.

@navnit-elastic, yes please. The idea is to provide better UX to the users, so updating the navigation panel when new dashboards are added is part of it. Please update any existing navigation panels with this new dashboard link.

kcreddy reviewed (commented):
Please add navigation panel and update the screenshot.
https://github.com/elastic/integrations/pull/13364#issuecomment-2772310032

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/24274)
* Commit: 5c29b2d20a40e9f10cc136e511b12aaf27dcfc0d

### History
* :green_heart: [Build #24204](https://buildkite.com/elastic/integrations/builds/24204) succeeded db506d4d7107277a204422c54ab379e42d1ae610
* :green_heart: [Build #24189](https://buildkite.com/elastic/integrations/builds/24189) succeeded 559379f73866f69ecb4f349a0e1a3d5d6f536f37
* :broken_heart: [Build #24188](https://buildkite.com/elastic/integrations/builds/24188) failed 590226cd41ba4ad1f3dc01a1d46eaaa09f85192e
* :green_heart: [Build #24104](https://buildkite.com/elastic/integrations/builds/24104) succeeded b2db8ff0bb0255d51ce8573d2c2ed0de87d7c7e5

cc @navnit-elastic

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/24274","number":24274,"commit":"5c29b2d20a40e9f10cc136e511b12aaf27dcfc0d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/24204","number":24204,"commit":"db506d4d7107277a204422c54ab379e42d1ae610"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/24189","number":24189,"commit":"559379f73866f69ecb4f349a0e1a3d5d6f536f37"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/24188","number":24188,"commit":"590226cd41ba4ad1f3dc01a1d46eaaa09f85192e"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/24104","number":24104,"commit":"b2db8ff0bb0255d51ce8573d2c2ed0de87d7c7e5"}],"number":24274}
buildkite-pr-comment-->
